ansible_base/resource_registry/apps.py

```diff
@@ -31,6 +31,8 @@ def initialize_resources(sender, **kwargs):
     if apps is None:
         from django.apps import apps
 
+    from ansible_base.resource_registry.models import init_resource_from_object
+
     Resource = apps.get_model("dab_resource_registry", "Resource")
```

Comment:
Just thoughts here:

You should be free to import `Resource` from models when this is called. What you _actually need_ is a heuristic to know if the migration is complete. Practically you only need this for `dab_resource_registry`. That's why moving out of the `@classmethod` is not hitting directly at the issue you're trying to fix.

this is code from AWX which does that

https://github.com/ansible/awx/blob/devel/awx/main/middleware.py#L203-L205

I wish there was something more direct, like a parameter from the signal.

Reply:
Interesting, I'll look into this. I tried importing resource directly, but it complained that ContentType was invalid.

Reply:
I've worried about DAB post_migrate signals having ordering problems relative to permissions / contenttypes, but so far I have never seen that happen. The resource registry does list contenttypes migration as a dependency so it's not an ordering problem on that layer.